pkg/skaffold/deploy/kpt.go

```diff
@@ -47,3 +56,53 @@ func (k *KptDeployer) Cleanup(ctx context.Context, out io.Writer) error {
 func (k *KptDeployer) Render(ctx context.Context, out io.Writer, builds []build.Artifact, offline bool, filepath string) error {
 	return nil
 }
+
+// getApplyDir returns the path to applyDir if specified by the user. Otherwise, getApplyDir
+// creates a hidden directory in place of applyDir.
+func (k *KptDeployer) getApplyDir(ctx context.Context) (string, error) {
+	if k.ApplyDir != "" {
+		if _, err := os.Stat(k.ApplyDir); os.IsNotExist(err) {
+			return "", err
+		}
+		return k.ApplyDir, nil
+	}
+
+	applyDir := ".kpt-hydrated"
+
+	// 0755 is a permission setting where the owner can read, write, and execute.
+	// Others can read and execute but not modify the file.
+	if err := os.MkdirAll(applyDir, 0755); err != nil {
+		return "", fmt.Errorf("applyDir was unspecified. creating applyDir: %w", err)
+	}
+
+	cmd := exec.CommandContext(ctx, "kpt", kptCommandArgs(applyDir, []string{"live", "init"}, nil, nil)...)
```

Comment:
We only call `kpt live init` if the inventory-template.yaml doesn't exist in the applyDir.

Reply:
I agree. The reason I was hesitant with checking if it doesn't exist first is because users can change the name of inventory-template.yaml and `kpt live init` will end up creating another template resource.